plugins/vendor/lib/model/VendorIntegrationPeer.php

```diff
@@ -109,4 +109,39 @@ public static function retrieveSingleVendorByPartner($partnerId, $vendorType)
 		return self::doSelectOne($c);
 	}
 
+	/**
+	 * @param $partnerId
+	 * @param $accountID
+	 * @param $vendorType
+	 * @return VendorIntegration
+	 * @throws PropelException
+	 */
+	public static function retrieveSingleVendorByAccountAndPartner($partnerId, $accountID, $vendorType)
```

Comment:
instead of duplicating the functions add optional partner id param to the functions retrieveSingleVendorPerPartnerNoFilter and retrieveSingleVendorPerPartner

Reply:
done